### PR TITLE
fix(shrexeds): hotfixes for respecting context

### LINKED ISSUE #2110: getters/shrex: GetEDS does not respect parent context deadline in all cases
There is an assumption inside of GetEDS that the call to `Peer` will check context for a deadline. This is not always the case, and we should be explicit about checking contexts anyways.

### PR DESCRIPTION
**Pull request by distractedm1nd**
Closes #2110, related to #2109.
Only adds a hotfix for 2109

### CODE REVIEW

File: share/p2p/shrexeds/client.go [hunk 1]
```diff
@@ -45,29 +45,46 @@ func (c *Client) RequestEDS(
...
+			log.Warnw("client: eds request to peer failed",
+				"peer", peer,
+				"hash", dataHash.String(),
+				"err", err)
+		}
+
+		return nil, err
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
```
Comment (walldiss) [inline share/p2p/shrexeds/client.go:87]: Ideally this context should have higher deadline than original one and log and error if it happen to timeout before `doRequest` call returns. Also would be nice to reset the stream here to enforce network call cancelation

File: share/getters/shrex.go [hunk 2]
```diff
@@ -67,6 +67,11 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 		err     error
 	)
 	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
```
Comment (walldiss) [inline share/getters/shrex.go:74]: why use select on single channel when you could check `ctx.Err != nil`?

File: share/p2p/shrexeds/client.go [hunk 3]
```diff
@@ -45,29 +45,46 @@ func (c *Client) RequestEDS(
...
+	// this is a hotfix for something inside of doRequest not respecting context.
+	// should be reverted when closing issue #2109
+	go func() {
+		eds, err := c.doRequest(ctx, dataHash, peer)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		resultChan <- eds
+	}()
```
Comment (renaynay) [inline share/p2p/shrexeds/client.go:60]: see my dm

**Comment by distractedm1nd** [comment]
closing and reopening because gh broke diffs

**Pull request closed without merging**